File: feed.py
import pandas as pd

# 讀取 & 清洗資料
chicks = pd.read_csv("chicks.csv")
chicks = chicks.dropna(subset=["weight"])
feed_options = chicks["feed"].unique()

# 印出選項選單
for i, option in enumerate(feed_options, start=1):
    print(f"{i}. {option}")

# 使用者選擇（避免使用 input 作為變數名稱，以免覆蓋 Python 內建函式）
user_input = input("Please Choose one option: ")

# 嘗試將字串轉為整數並驗證範圍
try:
    feed_choice = int(user_input)
    # 判斷輸入是否超出選單範圍（注意 Python 索引為 0-based，因此選單顯示 1~N，對應索引為 feed_choice - 1）
    if 1 <= feed_choice <= len(feed_options):
        selected_choice = feed_options[feed_choice - 1]
        print(chicks[chicks["feed"] == selected_choice])
    else:
        print("輸入非法")
except ValueError:
    print("輸入非法")

File: feed.R
#读取&清洗数据
chicks <- read.csv("chicks.csv")
chicks <- subset(chicks, !is.na(weight))
feed_options <- unique(chicks$feed)

formatted_options <- paste0(1:length(feed_options), ". ", feed_options)
cat(formatted_options, sep = "\n")
#用户选择
input <- readline("Please Choose one option: ")
# 嘗試將字串轉為整數（使用 suppressWarnings 隱藏轉型失敗時的警告訊息）
feed_choice <- suppressWarnings(as.integer(input))
# 判斷輸入是否非法：是否為 NA，或者是否超出選單範圍
if (
    is.na(feed_choice) || feed_choice < 1 || feed_choice > length(feed_options)
) {
    cat("輸入非法\n")
} else {
    selected_choice <- feed_options[feed_choice]
    print(subset(chicks, feed == selected_choice))
}
